lib/block_view.go

```diff
@@ -7060,6 +7342,153 @@ func (bav *UtxoView) _connectSwapIdentity(
 	return totalInput, totalOutput, utxoOpsForTxn, nil
 }
 
+// _verifyAccessSignature verifies if the accessSignature is correct. Valid
+// accessSignature is the signed hash of (derivedPublicKey + expirationBlock)
+// in DER format, made with the ownerPublicKey.
+func _verifyAccessSignature(ownerPublicKey []byte, derivedPublicKey []byte,
+	expirationBlock uint64, accessSignature []byte) error {
+
+	// Compute a hash of derivedPublicKey+expirationBlock
+	expirationBlockByte := UintToBuf(expirationBlock)
```

Comment:
Nit: Did you mean for this to be singular "byte" rather than plural "bytes"? UintToBuf returns a slice not a single byte.